### ISSUE TITLE
[Exceptions] Revise logic of `try`

### ISSUE THREAD

**Issue by drkameleon** [arturo-lang/arturo#1749]
**Describe the bug**

So, let's say we want to perform some not-so-safe code.

We have `try`. Perfect.
The code will run and if an exception is thrown, it will return the Error value. If everything goes fine, it will return `null`.

So...
let's do:

```
loop 10000 [
    try [
         ; something
    ]
]
```

and  we'll end up with 10K nulls on the stack. Yay! How awesome! I know, we'd have to set it to a variable - like `whatever: try` (looks very nice - not) -  or check for the error every time, even though I may not want to...

And then I remember we have `throws?`... This must be working. Right? Wrong!
This one apparently works **identically**, although this time it returns true/false (I get it we don't get the error; we get IF there was an error).

Long story short: this has to be fixed ASAP.

**Proposals**

- remove `throws?` altogether
- `try` should return something truthful/non-null if the block succeeded - not the other way around. I mean `unless try` means to me "if it didn't work", not if "it did work"
- perhaps we could add an option to actually get the result above (like e.g. `check`?)... so `try []` wouldn't return anything at all and `try.check []` would return Error if failed (this should be equivalent to NULL in all conditionals) and True if it succeeds
- perhaps we could add a new command - and one (e.g. `try`) would mean "do safely" and the other one would mean "do safely and return result"
